### ISSUE TITLE
TalkBack cannot select UI elements after `Gesture.Pan()` triggers

### ISSUE THREAD

**Issue by tpcstld** [software-mansion/react-native-gesture-handler#3407]
### Description

After `Gesture.Pan()` is triggered. TalkBack will no longer be able to focus UI elements using the press-and-hold gesture. TalkBack is still able to select elements via the swipe-left and swipe-right gesture. This issue happens on both the new and old architecture.

https://github.com/user-attachments/assets/286c9bf9-1eac-4666-a981-b4793c58da19

### Steps to reproduce

Repro:

```tsx
import * as React from "react";
import { Button, Text, View } from "react-native";
import {
  Gesture,
  GestureDetector,
  GestureHandlerRootView,
} from "react-native-gesture-handler";

function useGesture() {
  return React.useMemo(() => {
    return Gesture.Pan();
  }, []);
}

export default function HomeScreen() {
  const gesture = useGesture();

  return (
    <GestureHandlerRootView>
      <GestureDetector gesture={gesture}>
        <View
          style={{ flex: 1, alignItems: "center", justifyContent: "center" }}
        >
          <Text>This is some selectable text.</Text>
          <Button title="And this is a button" />
        </View>
      </GestureDetector>
    </GestureHandlerRootView>
  );
}
```

1. `npm run android` to build and launch the app on a physical device.
2. Press and hold over the text and button to confirm that it is focusable on talkback.
3. Do a 2 finger swipe up (or down) gesture to trigger the `Gesture.Pan`. Note that the gesture is not actually connected to anything.
4. Press and hold over the text and button again.

You should observe that step 4 fails to work.

### Snack or a link to a repository

https://github.com/tpcstld/rn-stuff/tree/tpcstld/pan-gesture-bug

### Gesture Handler version

2.23.1

### React Native version

0.77.1

### Platforms

Android

### JavaScript runtime

Hermes

### Workflow

None

### Architecture

Paper (Old Architecture)

### Build type

Debug mode

### Device

Real device

### Device model

Pixel 7

### Acknowledgements

Yes

**Comment by tpcstld** [comment]
From my investigation, the following patch seems to fix it:

```diff
diff --git a/android/src/main/java/com/swmansion/gesturehandler/core/GestureHandlerOrchestrator.kt b/android/src/main/java/com/swmansion/gesturehandler/core/GestureHandlerOrchestrator.kt
index ebcbb0dcd47a01781eb78728daa942edd3313eee..8028e73f1ffdd01c9a043657cae5d18f40b6b7b0 100644
--- a/android/src/main/java/com/swmansion/gesturehandler/core/GestureHandlerOrchestrator.kt
+++ b/android/src/main/java/com/swmansion/gesturehandler/core/GestureHandlerOrchestrator.kt
@@ -6,6 +6,7 @@ import android.view.MotionEvent
 import android.view.View
 import android.view.ViewGroup
 import android.widget.EditText
+import com.swmansion.gesturehandler.react.RNGestureHandlerRootHelper
 import java.util.*
 import kotlin.collections.HashSet
 
@@ -481,7 +482,7 @@ class GestureHandlerOrchestrator(
           }
 
           // we don't want to extract gestures other than hover when processing hover events
-          if (event.action in listOf(MotionEvent.ACTION_HOVER_EXIT, MotionEvent.ACTION_HOVER_ENTER, MotionEvent.ACTION_HOVER_MOVE) && handler !is HoverGestureHandler) {
+          if (event.action in listOf(MotionEvent.ACTION_HOVER_EXIT, MotionEvent.ACTION_HOVER_ENTER, MotionEvent.ACTION_HOVER_MOVE) && handler !is HoverGestureHandler && handler !is RNGestureHandlerRootHelper.RootViewGestureHandler) {
             continue
           }
 
diff --git a/android/src/main/java/com/swmansion/gesturehandler/react/RNGestureHandlerRootHelper.kt b/android/src/main/java/com/swmansion/gesturehandler/react/RNGestureHandlerRootHelper.kt
index b62407baa623f3a61f0da23dfb80186cc1e93fd2..96bbf3cadc437d64156c965a201252de33f25b9e 100644
--- a/android/src/main/java/com/swmansion/gesturehandler/react/RNGestureHandlerRootHelper.kt
+++ b/android/src/main/java/com/swmansion/gesturehandler/react/RNGestureHandlerRootHelper.kt
@@ -71,6 +71,19 @@ class RNGestureHandlerRootHelper(private val context: ReactContext, wrappedView:
       }
     }
 
+    override fun onHandleHover(event: MotionEvent, sourceEvent: MotionEvent) {
+      val currentState = state
+      // we shouldn't stop intercepting events when there is an active handler already, which could happen when
+      // adding a new pointer to the screen after a handler activates
+      if (currentState == STATE_UNDETERMINED && (!shouldIntercept || orchestrator?.isAnyHandlerActive() != true)) {
+        begin()
+        shouldIntercept = false
+      }
+      if (event.actionMasked == MotionEvent.ACTION_HOVER_EXIT) {
+        end()
+      }
+    }
+
     override fun onCancel() {
       shouldIntercept = true
       val time = SystemClock.uptimeMillis()
```

I can put this up for PR, but I don't have much experience with the internals of `react-native-gesture-handler` and so I have _no idea_ if this is architecturally sound.

The root cause seems to be that TalkBack leverages the hover actions, and in https://github.com/software-mansion/react-native-gesture-handler/pull/2455/files#diff-5fd8227470d0bf0e7bdeccf7715263f54f1ccadc7d873c701c4e89d1eae869c0R478 we introduced considerable handling for hover actions for the hover gesture.

Part of the changes involved _not_ handling `ACTION_HOVER_FOO` events specially, and this difference in handling meant that the `RNGestureHandlerRootHelper` did not really handle on these hover events. This meant that [`shouldIntercept`](https://github.com/software-mansion/react-native-gesture-handler/blob/3e21b6c2e27b18353589067293dfad7be3a14f9e/android/src/main/java/com/swmansion/gesturehandler/react/RNGestureHandlerRootHelper.kt#L104) would not be cleared when we received a hover event after a gesture has been activated and finished.

**Comment by m-bert** [comment]
Hi @tpcstld! Thank you for investigating this issue! 

> I can put this up for PR.

Of course, it would be great! 

> I don't have much experience with the internals of react-native-gesture-handler and so I have no idea if this is architecturally sound.

The logic itself is fine, but it should be a bit refactored. First of all `onHandleHover` does exactly the same what `onHandle`, so it can be extracted into separate function. Also this first `if` condition (in `GestureHandlerOrchestrator.kt`) is now quite big, maybe we should also consider making it a function with appropriate name 😅 

I'll make sure to review it once your PR is ready 😄
